components/rhel/rhel-ose/Vagrantfile

```diff
@@ -1,65 +1,68 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
-# Usage:
-#
-# To use this Vagrantfile, you(optional):
-# * Connected to internal Red Hat network if you need to any package update
-# * Have 'vagrant-registration' plugin installed
-# * Have valid RH employee subscription account
-#
-# The public IP address the VM created by Vagrant will get.
-# You will use this IP address to connect to OpenShift web console.
+# The private network IP of the VM. You will use this IP to connect to OpenShift.
 PUBLIC_ADDRESS="10.1.2.2"
 
-#check if vagrant-service-manager is installed on system
-unless Vagrant.has_plugin?("vagrant-service-manager")
-  raise "vagrant-service-manager plugin is not installed, run `vagrant plugin install vagrant-service-manager` to install the plugin."
-end
+# Number of virtualized CPUs
+VM_CPU = ENV['VM_CPU'] || 2
+
+# Amount of available RAM
+VM_MEMORY = ENV['VM_MEMORY'] || 3072
+
+# Validate required plugins
+REQUIRED_PLUGINS = %w(vagrant-service-manager vagrant-registration)
+message = ->(name) { "#{name} plugin is not installed, run `vagrant plugin install #{name}` to install it." }
+errors = []
+REQUIRED_PLUGINS.each {|plugin| errors << message.call(plugin) unless Vagrant.has_plugin?(plugin) }
+msg = errors.size > 1 ? "Errors: \n* #{errors.join("\n* ")}" : "Error: #{errors.first}"
+raise Vagrant::Errors::VagrantError.new, msg unless errors.size == 0
 
 Vagrant.configure(2) do |config|
   config.vm.box = "cdkv2"
 
-  config.registration.username = ENV['SUB_USERNAME']
-  config.registration.password = ENV['SUB_PASSWORD']
-
   config.vm.provider "virtualbox" do |v, override|
-    v.memory = 1024
-    v.cpus   = 2
+    v.memory = VM_MEMORY
+    v.cpus   = VM_CPU
     v.customize ["modifyvm", :id, "--ioapic", "on"]
     v.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
   end
 
   config.vm.provider "libvirt" do |v, override|
+    v.memory = VM_MEMORY
+    v.cpus   = VM_CPU
     v.driver = "kvm"
-    v.memory = 1024
-    v.cpus   = 2
   end
 
   config.vm.network "private_network", ip: "#{PUBLIC_ADDRESS}"
 
+  if ENV.has_key?('SUB_USERNAME') && ENV.has_key?('SUB_PASSWORD')
+    cdk.registration.username = ENV['SUB_USERNAME']
+    cdk.registration.password = ENV['SUB_PASSWORD']
+  end
+
   config.vm.provision "shell", inline: <<-SHELL
-    sudo systemctl start openshift
-    sudo systemctl enable openshift
+    systemctl enable openshift
+    systemctl start openshift
```

Comment:
@bexelbie This PR for CDK vagrant file changes and I don't want to have `sccli` there because it does again try to pull images (even it is there already as cached) which will increase startup time of service by a min or more. We need a way for sccli to figure out if image already exist then don't try pulling it so we can avoid this extra time.